Update the minimum engineering team size guideline (#3914)
sourcegraph/about: handbook/engineering/eng_org.md

- A product manager
- An engineering manager
- Appropriate product design support depending on the work of the team

> - Four engineers, with a minimum of two per skillset (e.g. 2 backend and 2 frontend, or 4 backend/frontend)

Reviewer: ```suggestion
- Four engineers, with a minimum of two per skillset (examples: 2 backend + 2 frontend, 4 backend, 4 frontend)
```
